mm2src/coins/rpc_command/get_estimated_fees.rs

```diff
@@ -0,0 +1,217 @@
+                                       // To support fee estimations for other chains add a FeeEstimatorContext for a new chain
+
+#[derive(Debug, Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum FeeEstimatorError {
+    #[display(fmt = "Coin not activated")]
+    CoinNotActivated,
+    #[display(fmt = "Gas fee estimation not supported for this coin")]
+    CoinNotSupported,
+    #[display(fmt = "Chain id not supported")]
+    ChainNotSupported,
+    #[display(fmt = "Gas fee estimator is already started")]
+    AlreadyStarted,
+    #[display(fmt = "Transport error: {}", _0)]
+    Transport(String),
+    #[display(fmt = "Gas fee estimator is not running")]
+    NotRunning,
+    #[display(fmt = "Internal error: {}", _0)]
+    InternalError(String),
+}
+
+impl HttpStatusCode for FeeEstimatorError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            FeeEstimatorError::CoinNotActivated
+            | FeeEstimatorError::CoinNotSupported
+            | FeeEstimatorError::ChainNotSupported
+            | FeeEstimatorError::AlreadyStarted
+            | FeeEstimatorError::NotRunning => StatusCode::BAD_REQUEST,
+            FeeEstimatorError::Transport(_) | FeeEstimatorError::InternalError(_) => StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
+}
+
+impl From<NumConversError> for FeeEstimatorError {
+    fn from(e: NumConversError) -> Self { FeeEstimatorError::InternalError(e.to_string()) }
+}
+
+impl From<String> for FeeEstimatorError {
+    fn from(e: String) -> Self { FeeEstimatorError::InternalError(e) }
+}
+
+/// Gas fee estimator loop context,
+/// runs a loop to estimate max fee and max priority fee per gas according to EIP-1559 for the next block
+///
+/// This FeeEstimatorContext handles rpc requests which start and stop gas fee estimation loop and handles the loop itself.
+/// FeeEstimatorContext maintains a list of eth coins or tokens which connected and use the estimator.
+/// The loop estimation starts when first eth coin or token calls the start rpc and stops when the last coin or token, using it, calls the stop rpc.
+/// FeeEstimatorContext keeps the latest estimated gas fees in the context and returns them as rpc response
+pub struct FeeEstimatorContext {
+    estimated_fees: Arc<AsyncMutex<FeePerGasEstimated>>,
+    abort_handler: AsyncMutex<Option<AbortOnDropHandle>>,
+}
+
+impl FeeEstimatorContext {
+    fn new() -> Result<Self, String> {
+        Ok(Self {
+            estimated_fees: Default::default(),
+            abort_handler: AsyncMutex::new(None),
+        })
+    }
+
+    fn from_ctx(ctx: MmArc) -> Result<Arc<FeeEstimatorContext>, String> {
+        Ok(try_s!(from_ctx(&ctx.fee_estimator_ctx, move || {
+            FeeEstimatorContext::new()
+        })))
+    }
+
+    /// Fee estimation update period in secs, basically equals to eth blocktime
+    const fn get_refresh_interval() -> f64 { 15.0 }
```

Comment:
If multiple coins are supported according to the previous comment, this should be block time, we have this in coin config called `avg_blocktime`. It might be removed anyway if we relied on `eth_subscribe` to get notifications for new headers/blocks.

Reply:
yes I also was thinking about syncing fee updates with new block coming, thank you.
Though some gas fee providers provide more frequent and dynamic updates than each block, based on current mempool changes. For now I think we can stick with blocktime

Reply:
> Though some gas fee providers provide more frequent and dynamic updates than each block, based on current mempool changes.

Please add a checklist in the original issue with this item and other items that we decide to do later or in next PRs. This will be a reminder to not close the issue until all things are resolved.